### PR TITLE
Move config fragments for contianers, add workaround for overlayfs/IMA deadlock.

### PR DESCRIPTION
**Pull request by klihub**
Moved kernel configuration fragments for systemd-nspawned containers to meta-ostro-bsp.
Added a patch/workaround for preventing a kernel deadlock when writing through overlayfs to a file on an IMA-protected filesystem.
Added kernel configuration fragments for enabling lockdep and a debuggable kernel.

### CODE REVIEW

File: recipes-kernel/linux-yocto/linux-yocto/lockdep.cfg [hunk 1]
```diff
@@ -0,0 +1,99 @@
+CONFIG_EXPERT=y
+CONFIG_KALLSYMS_ALL=y
+# CONFIG_DEBUG_PERF_USE_VMALLOC is not set
```
Comment (jlaako) [inline recipes-kernel/linux-yocto/linux-yocto/lockdep.cfg]: If these options are not intentionally to be specifically turned off, you can remove this to leave them as-is.

Comment (klihub) [inline recipes-kernel/linux-yocto/linux-yocto/lockdep.cfg]: Redone with all settings off dropped (like you pointed out, I don't think we need to force any of them to be off). Boot-tested a resulting kernel and at least the necessary lockdep bits stayed on and were available.

File: recipes-kernel/linux-yocto/linux-yocto_4.4.bbappend [hunk 2]
```diff
@@ -170,3 +170,24 @@ SRC_URI_append_intel-corei7-64 = " \
     file://nspawn-firewall.cfg \
     file://nspawn-devpts.cfg \
 "
+
+# Overlayfs deadlock fix.
+SRC_URI_append = " \
```
Comment (jlaako) [inline recipes-kernel/linux-yocto/linux-yocto_4.4.bbappend]: This generic append should be removed...

File: recipes-kernel/linux-yocto/linux-yocto_4.4.bbappend [hunk 3]
```diff
@@ -170,3 +170,24 @@ SRC_URI_append_intel-corei7-64 = " \
...
+    file://0001-ovl-setxattr-don-t-deadlock-when-called-from-ima_fix.patch \
+"
+
+SRC_URI_append_edison = " \
```
Comment (jlaako) [inline recipes-kernel/linux-yocto/linux-yocto_4.4.bbappend]: This patch won't apply for Edison kernel due to lack of overlayfs.

File: recipes-kernel/linux-yocto/linux-yocto_4.4.bbappend [hunk 4]
```diff
@@ -131,3 +131,14 @@ SRC_URI_append = " file://ecryptfs.cfg"
 # Backport ambient capabilities support
 SRC_URI_append_edison = " file://0001-edison-capabilities-ambient-capabilities.patch"
 SRC_URI_append_edison = " file://0002-edison-capabilities-add-a-securebit-to-disable-PR_CAP_AMBIE.patch"
+
+# Backport AK8975 iio device name support
+SRC_URI_append_edison = " file://0001-iio-ak8975-Add-device-name.patch"
```
Comment (jlaako) [inline recipes-kernel/linux-yocto/linux-yocto_4.4.bbappend]: Hmmh, where did this come from to this diff? I guess this is unintended side-effect?

**Pull request merged**